### DISCUSSION TITLE
Offline maps using MBTiles files

### DISCUSSION THREAD

**Discussion by sjchristi** [maplibre/maplibre-native#393] [General]
Hello Maplibre,

I'm building a mobile (ios/android) app that is often used while users are offline.  We have generated `.mbtiles` files for regions all over the world, split into countries (or for large countries, into smaller regions -- like states / provinces / etc).  Users download the regions they need when they will be offline, and then the map uses the mbtiles files to render the map they are in.

Currently, to render a `.mbtiles` file, we setup a `source` in the style `sources` that looks like this:

```js
      newMapStyle.sources.openmaptiles = {
        type: 'vector',
        //tiles: [`mbtiles://${fallback.path}`],
        url: `mbtiles://file://${mbTilePath}`,
      };
```

I had some questions about the above code.  First, on iOS, we could use the `tiles:` section above to point to the map -- but that didn't work on Android.  However, when we used the `url:` pattern above, it works on both iOS and Android.  Is there any reason for that, or why one should be preferred over the other, or is that even a bug that Android doesn't support the `tiles:` syntax?

Second - when we do this, we are only able to specify a single `.mbtiles` file.  It would be ideal for us if we could actually specify multiple `.mbtiles` sources somehow.  For example, suppose a user download multiple countries/states that are next to each other -- ideally maplibre would just search all of them for the tile before giving up.  Is that possible somehow right now?

To workaround the problem above right now, we look at what the user is looking at and get the center lat,lng for the map, and then figure out which offline region should have that tile, and then update the style based on that..  Does this seem reasonable, or is there a better/different way?

(side note: would be happy to share more sample code if it would be helpful)

thanks so much,
Sam

**Comment by M0das** [comment]
Hello, Sorry for replying In this discussion 4 years later :), I'm a react-native dev (very noob one) and trying to build an offline map viewer tool without any connection to internet (I later want to use it for drawing routes for a speceific reasons), and now I'm trying to understand how MapLibre and .mbtiles and .pbf files work, I've got my city's .mbtiles and .osm.pbf file and I have style.json from OSM to style the tiles, but I cant seem to make the app show the map, If you have a full structure to make the map showing on the screen please share it so I have a better idea, and If you have tips please share.

thanks, 
modas

**Reply by louwers** [reply]
Hey @M0das. OP in this thread seems to want to do something more complicated than your question.

In your case you should edit the style and use the `mbtiles:///path/to/file.mbtiles` as the source. Make sure the MBTiles file is available on the file system.

If you want a React Native example it may be a good idea to make an issue there if the documentation doesn't cover it.
